test/support/page_objects/common.js

```diff
@@ -299,11 +299,20 @@ export default class Common {
     return exists;
   }
 
-  findTestSubject(selector) {
+  findTestSubject(selector, timeout = defaultFindTimeout) {
     this.debug('in findTestSubject: ' + testSubjSelector(selector));
+    let originalFindTimeout = null;
```

Comment:
This would probably be a lot cleaner with `async`/`await`

```js
  async findTestSubject(selector, timeout = defaultFindTimeout) {
    this.debug('in findTestSubject: ' + testSubjSelector(selector));
    const originalFindTimeout = await this.remote.getFindTimeout();
    try {
      return await this.remote
        .setFindTimeout(timeout)
        .findDisplayedByCssSelector(testSubjSelector(selector))
    } finally {
      this.remote.setFindTimeout(originalFindTimeout)
    }
  }
```